libsupport/include/katana/Cache.h

```diff
@@ -53,41 +53,33 @@ struct CacheStats {
   uint64_t insert_hit_count{0ULL};
 };
 
-template <typename Key, typename Value, typename CallerPointer = void*>
+template <typename Value>
 class KATANA_EXPORT Cache {
+  using Key = katana::Uri;
   using ListType = std::list<Key>;
   struct MapValue {
     Value value;
     // This allows us to delete the old position in the LRU list without a scan
     typename ListType::iterator lru_it;
```

Comment:
So an iterator is used as an explicit pointer to a position in the doubly linked list correct? I was wondering why we didn't use something like an index or pointer, but this looks a lot cleaner. Very interesting

Reply:
I like it.  I coped the design from boost.
`https://www.boost.org/doc/libs/1_67_0/boost/compute/detail/lru_cache.hpp`